### PR TITLE
Allow specifying a particular resource name

### LINKED ISSUE #13: Allow specifying a particular resource name
Specify a particular resource e.g. `kubectl-who-can use pod-security-policy <name>`

### PR DESCRIPTION
**Pull request by danielpacak**
Resolves #13

### CODE REVIEW

File: cmd/list_test.go [hunk 1]
```diff
@@ -110,3 +110,90 @@ func TestWhoCan_ValidateNamespace(t *testing.T) {
...
+		verb         string
+		resource     string
+		resourceName string
+
+		rule rbacv1.PolicyRule
+
+		matches bool
+	}{
+		{
+			name: "scenario1",
```
Comment (lizrice) [inline cmd/list_test.go]: Does the command work for singular and plural forms? For example "get pods", "get pod", "get pods foo" and "get pod foo". If it works already, please could you add some test cases to cover those options? If it needs extra work  an issue / PR would be fine.

File: cmd/root.go [hunk 2]
```diff
@@ -14,44 +14,32 @@ import (
...
 		if len(args) < 2 {
 			return errors.New("please specify at least a verb and a resource type")
 		}
-		if len(args) > 2 {
-			return errors.New("TODO! resource name checking not yet implemented")
-		}
 		return nil
 	},
 	Run: func(cmd *cobra.Command, args []string) {
-		if err := cmd.ParseFlags(args); err != nil {
```
[the remark this answers is not in the document]
Comment (danielpacak) [reply cmd/root.go]: I'm pretty sure it's redundant here as the Cobra module calls `ParseFlags` implicitly from the`RootCmd.Execute()` stack:

ParseFlags

What's more the `args` to this anonymous function are stripped from the `os.Args` corresponding to the flags (i.e. `["-n", "foo"]`).

[the remark this answers is not in the document]
Comment (lizrice) [reply cmd/root.go]: Oh cool!

File: cmd/list_test.go (hunk 1 above)
Comment (danielpacak) [reply cmd/list_test.go]: Nope. Currently we only use stings comparison so `get pod foo` won't work.  I'd take it as a dedicated issue / PR to add also checks for resource existence, e.g.:

```
./kubectl-who-can get frogs
Warning: the server doesn't have a resource type 'frog'
```

BTW  I realized that`kubectl auth can-i` supports wildcards:
```
kubectl auth can-i '*' pods --as=dpacak
```

Do you think it makes sens to add it to who-can as well?

File: cmd/list.go [hunk 3]
```diff
@@ -91,46 +127,62 @@ func (w *whoCan) getClusterRoles() error {
...
 }
 
-func (w *whoCan) filterClusterRoles(rl *rbacv1.ClusterRoleList) {
-	// Only interested in ClusterRoles that relate to the verbs and resources we care about
-	for _, item := range rl.Items {
-		glog.V(3).Info(fmt.Sprintf("Cluster Role: %s", item.Name))
-		// Roles contain PolicyRules
-		for _, rule := range item.Rules {
-			w.policyRuleMatch(rule, item.Name, true)
+func (w *whoCan) policyMatches(rule rbacv1.PolicyRule) bool {
```
[the remark this answers is not in the document]
Comment (danielpacak) [reply cmd/list.go]: Done

File: cmd/list_test.go (hunk 1 above)
Comment (lizrice) [reply cmd/list_test.go]: Yes to both!

**Pull request merged**